tokio/src/runtime/context.rs

```diff
@@ -146,37 +152,56 @@ cfg_rt! {
     }
 
     pub(crate) fn try_enter_blocking_region() -> Option<BlockingRegionGuard> {
-        ENTERED.with(|c| {
-            if c.get().is_entered() {
+        CONTEXT.try_with(|c| {
+            if c.runtime.get().is_entered() {
                 None
             } else {
                 Some(BlockingRegionGuard::new())
             }
-        })
+            // If accessing the thread-local fails, the thread is terminating
+            // and thread-locals are being destroyed. Because we don't know if
+            // we are currently in a runtime or not, we default to being
+            // permissive.
```

Comment:
The test that failed tested storing a runtime in a thread-local. I think the test just happened to pass before due to the order in which thread-local variables were destroyed. This is a behavior change, but I think it is necessary, otherwise shutting down a runtime from a thread-local can double panic and abort the process.

Reply:
Being permissive here seems good to me.

Reply:
I have no objections here.